### PR TITLE
chore: assign MSI to AKS Windows build VMs

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Assign MSI to AKS Windows build VMs so it can use MSI to download private packages.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/configure-windows-vhd.ps1 [hunk 1]
```diff
@@ -211,14 +211,22 @@ function Get-PrivatePackagesToCacheOnVHD {
...
+        $env:AZCOPY_MSI_RESOURCE_STRING=$env:WindowsMSIResourceString
+
         $urls = $env:WindowsPrivatePackagesURL.Split(",")
         foreach ($url in $urls) {
             $fileName = [IO.Path]::GetFileName($url.Split("?")[0])
             $dest = [IO.Path]::Combine($dir, $fileName)
 
             Write-Log "Downloading a private package to $dest"
-            DownloadFileWithRetry -URL $url -Dest $dest -redactUrl
+            .\azcopy\*\azcopy.exe copy $URL $dest
```
Comment (AbelHu) [inline vhdbuilder/packer/configure-windows-vhd.ps1:226]: It is ok for now. We can add retry if it fails later.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

File: vhdbuilder/packer/configure-windows-vhd.ps1 [hunk 2]
```diff
@@ -211,14 +211,22 @@ function Get-PrivatePackagesToCacheOnVHD {
         $dir = "c:\akse-cache\private-packages"
         New-Item -ItemType Directory $dir -Force | Out-Null
 
+        Write-Log "Downloading azcopy"
+        Invoke-WebRequest -UseBasicParsing "https://aka.ms/downloadazcopy-v10-windows" -OutFile azcopy.zip
```
Comment (ShiqianTao) [inline vhdbuilder/packer/configure-windows-vhd.ps1:215]: Do you need to remove the azcopy.zip?

File: vhdbuilder/packer/configure-windows-vhd.ps1 (hunk 1 above)
Comment (junjiezhang1997) [reply vhdbuilder/packer/configure-windows-vhd.ps1:226]: It seems that azcopy has built-in retry for download https://github.com/search?q=repo%3AAzure%2Fazure-storage-azcopy%20MaxRetryPerDownloadBody&type=code

File: vhdbuilder/packer/configure-windows-vhd.ps1 (hunk 2 above)
Comment (junjiezhang1997) [reply vhdbuilder/packer/configure-windows-vhd.ps1:215]: Thanks for the comment. Added the remove step.

**Pull request merged**